src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -440,23 +446,70 @@ void activity_classifier::perform_training_iteration() {
     float batch_loss = std::accumulate(loss_batch.data(),
                                        loss_batch.data() + loss_batch.size(),
                                        0.f, std::plus<float>());
+
+    
     cumulative_batch_loss += batch_loss / batch.batch_info.size();
 
+    
+    const shared_float_array& output = results.at("output");
+    float cumulative_per_batch_accuracy = 0.f;
+    
+    for (size_t i = 0; i < batch.batch_info.size(); ++i) {
+      
+      shared_float_array output_chunk = output[i];
+
+      // Actual labels 
+      shared_float_array classes_chunk = batch.labels[i];
+      data_iterator::batch::chunk_info info = batch.batch_info[i];
+      
+
+      const float* output_ptr = output_chunk.data();
+      const float* truth_ptr = classes_chunk.data();
+      
+      size_t cumulative_samples = 0;
+      size_t seq_len =0;
+      float cumulative_seq_accuracy = 0.f;
+
+
+      while (cumulative_samples < info.num_samples) {
+
+        // Copy the probability vector to get accuracy.
+        std::copy(output_ptr, output_ptr + num_classes, preds.begin());
+        output_ptr += num_classes;
+        
+        cumulative_samples += prediction_window;
+        seq_len++;
+        truth_ptr ++;
+        if ((std::max_element(preds.begin(), preds.end())-preds.begin() == *truth_ptr ) ){
+          cumulative_seq_accuracy+=1.0;
+
+        };
+
+      }
+
+      cumulative_per_batch_accuracy += cumulative_seq_accuracy/ (seq_len + 1e-5);
+      
+    };
+
+    cumulative_batch_accuracy += cumulative_per_batch_accuracy/ (batch.batch_info.size() + 1e-5);
```

Comment:
Here too, let's not do the `+ 1e-5` bit. We already know that the batch should not be empty.